src/super_gradients/training/utils/callbacks.py

```diff
@@ -450,6 +448,34 @@ def __call__(self, context: PhaseContext):
         self.context = context
 
 
+class DetectionVisualizationCallback(PhaseCallback):
+    """
+    A callback that adds a visualization of a batch of detection predictions to context.sg_logger
+    Attributes:
+        freq: frequency (in epochs) to perform this callback.
+        batch_idx: batch index to perform visualization for.
+        classes: class list of the dataset.
+        last_img_idx_in_batch: Last image index to add to log. (default=-1, will take entire batch).
+    """
+    def __init__(self, phase: Phase, freq: int, post_prediction_callback: DetectionPostPredictionCallback, classes: list, batch_idx: int = 0, last_img_idx_in_batch: int = -1):
+        super(DetectionVisualizationCallback, self).__init__(phase)
+        self.freq = freq
+        self.post_prediction_callback = post_prediction_callback
+        self.batch_idx = batch_idx
+        self.classes = classes
+        self.last_img_idx_in_batch = last_img_idx_in_batch
+
+    def __call__(self, context: PhaseContext):
+        if context.epoch % self.freq == 0 and context.batch_idx == self.batch_idx:
+            # SOME CALCULATIONS ARE IN-PLACE IN NMS, SO CLONE THE PREDICTIONS
+            preds = (context.preds[0].clone(), None)
+            preds = self.post_prediction_callback(preds)
```

Comment:
but what if a DetectionPostPredictionCallback is applied on the predictions before this callback?
maybe add a warning in the documentation...

Reply:
This cannot be the case as we take the predictions from context.preds. 
In SgModel code we update the context with raw predictions always.

Reply:
great